### ISSUE TITLE
Wrong position when reloading one item

### ISSUE THREAD

**Issue by LPalmito** [BTMorton/angular2-grid#68]
Hey!

I've found an issue, following those steps (you can do it directly on this plunkr http://plnkr.co/edit/yDSSAhvOC7zFvhF5IT5M?p=preview):

- Update the first `widget` object (by clicking on the `reload widget #1` button)
- Drag and drop one of the widgets without moving its position (just click one `widget #1` for example)
-> the 2 widgets are going to change there respective positions even if that's not the behaviour we want

I've tried to find the root of the problem on my personal project, and it seems like it's coming from the `items` argument in the `updateWidgets` method. Here are my observations:

- Without reloading `widget #1`, the `items` array is the one we want
- When reloading `widget #1` before, the `items` array is reversed from what we want
- When trying this with more than 2 different widgets, the one updated "goes to the last index of the `items` array".

For example, if we have widgets A, B, C, D, E in that order. Before reloading, `items` is equal to `[info(A), info(B), info(C), info(D), info(E)]` and, if reloading widget X before:

- if X = A -> `items` is equal to `[info(B), info(C), info(D), info(E), info(A)]`
- if X = B -> `items` is equal to `[info(A), info(C), info(D), info(E), info(B)]`
- if X = C -> `items` is equal to `[info(A), info(B), info(D), info(E), info(C)]`
- if X = D -> `items` is equal to `[info(A), info(B), info(C), info(E), info(D)]`
- if X = E -> `items` is equal to `[info(A), info(B), info(C), info(D), info(E)]`

I don't really know how this `items` object is created but, is there a way to retrieve the "original info"? For example, instead of an array of `info(X)` having an array of `{X: info(X)}` so we don't care about the order and can use `X` as keys to retrieve the data we want instead?

By the way, thanks a lot for you're work, `angular2-grid` has been really helpful for my project so far!

**Comment by BTMorton** [comment]
I'm afraid it wouldn't really help you. In that situation, what you'd get would be:

* if X = A -> `items` is equal to `{1: info(B), 2: info(C), 3: info(D), 4: info(E), 5: info(A)}`
* if X = B -> `items` is equal to `{1: info(A), 2: info(C), 3: info(D), 4: info(E), 5: info(B)}`

etc.

The grid has no concept of what the item refers to in terms of the app. It only know what order they were added. When you do the reload of a single item, what it's doing is removing and then adding the item again, hence it being pushed to the last place. That's caused by the behaviour of the NgFor change detector.

I've updated your [plunkr](http://plnkr.co/edit/Yvx0QkyTdzEZTeswVmDz?p=preview) so that it doesn't swap the values and uses the dragStop/resizeStop events (itemChange was causing an infinite loop :S). It still puts the first item below the second one, but that's because it triggers a cascade after removing and then will adjust the position of the new item when adding to avoid conflicts. The alternative is to shove all the other items out of the way when adding. I can probably add in a config value to switch the behaviours around.
